xdsl/dialects/memref.py

```diff
@@ -379,6 +377,49 @@ class Subview(Operation):
 
     irdl_options = [AttrSizedOperandSegments()]
 
+    @staticmethod
+    def from_static_parameters(
+        source: SSAValue | Operation,
+        source_element_type: Attribute,
+        source_shape: Sequence[int],
+        offsets: Sequence[int],
+        sizes: Sequence[int],
+        strides: Sequence[int],
+    ) -> Subview:
+
+        source = SSAValue.get(source)
+
+        layout_strides = [1]
+        for input_size in reversed(source_shape[1:]):
+            layout_strides.insert(0, layout_strides[0] * input_size)
+
+        layout_offset = sum([
```

Comment:
```suggestion
        layout_offset = sum(
```
No need to construct intermediate list